### PR TITLE
eslint: add rule to detect erroneous external imports

### PR DESCRIPTION
**Pull request by iOvergaard**
This new rule ensures that we won't accidentally make imports from external libraries elsewhere. The only allowed places to do so are in tests, stories, and `src/external/*` modules.

We won't notice these errors in the dev-server, because Vite knows how to handle them, but it WILL become a problem in the compiled output used in the CMS.

BAD:
```js
import { html } from 'lit';
```

GOOD:
```js
import { html } from '@umbraco-cms/backoffice/external/lit';
```

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
 				"local-rules/umb-class-prefix": "error",
 				"local-rules/prefer-static-styles-last": "warn",
 				"local-rules/ensure-relative-import-use-js-extension": "error",
+				"local-rules/ensure-no-external-imports": "error",
```
Comment (madsrasmussen) [inline .eslintrc.json:48]: can we call the rule something like "app no direct node_module imports", so the name doesn't collide with the naming of our "external"-module?

**Review by madsrasmussen — CHANGES_REQUESTED** [review]
Really great to get this rule in place! You found some weird imports here an there. 

I only have one request regarding the rule name. It can be misunderstood because we call our own module "external".

File: .eslintrc.json (hunk 1 above)
Comment (iOvergaard) [reply .eslintrc.json:48]: Great idea! Here is what Github Copilot suggests, what do you think? 😅

> GitHub Copilot
You could consider naming the rule something like enforce-umbraco-external-imports. This name clearly communicates the purpose of the rule (to enforce imports from the @umbraco-cms/backoffice/external module) and should not conflict with the name of your external module.

Comment (madsrasmussen) [reply .eslintrc.json:48]: Brilliant 🎉 😄

**Review by nielslyngsoe — APPROVED** [review]
Looks good to me :-)

**Pull request merged**